projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -1062,10 +1062,11 @@ export class IgxTimePickerComponent extends PickerBaseDirective
 
     protected onStatusChanged() {
         if ((this._ngControl.control.touched || this._ngControl.control.dirty) &&
-            (this._ngControl.control.validator || this._ngControl.control.asyncValidator) &&
             !this._ngControl.disabled) {
-            if (this._inputGroup.isFocused) {
-                this.inputDirective.valid = this._ngControl.valid ? IgxInputState.VALID : IgxInputState.INVALID;
+            if (this._ngControl.control.validator || this._ngControl.control.asyncValidator) {
```

Comment:
Same as all the rest, try to simplify it and make sure we don't nest ternaries as they can be difficult to follow.